### PR TITLE
feat(core): sync site configs to KV for new protected app

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Sync site configs to KV for new protected app:

1. new library ProtectedAppsLibrary.
2. new method for KV operations in Cloudflare utils.
3. new system configs for KV.
4. Sync configs to Cloudflare KV when creating new protected application.

## Testing

Unit tests added.

No integration tests since we don't do test for outside services. There is no change in our side.

Local tested, when creating a new app, new KV record created:

image

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/libraries/protected-app.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+export type ProtectedAppLibrary = ReturnType<typeof createProtectedAppLibrary>;
+
+export const createProtectedAppLibrary = (queries: Queries) => {
+  const {
+    applications: { findApplicationById },
+  } = queries;
+
+  const syncAppConfigsToRemote = async (applicationId: string): Promise<void> => {
+    if (EnvSet.values.isIntegrationTest) {
```
Comment (simeng-li) [inline packages/core/src/libraries/protected-app.ts:15]: nit: Better add some comments.

File: packages/core/src/routes/applications/application.ts [hunk 2]
```diff
@@ -168,6 +166,12 @@ export default function applicationRoutes<T extends AuthedRouter>(
         ...rest,
       });
 
+      if (application.type === ApplicationType.Protected) {
+        await protectedApps.syncAppConfigsToRemote(application.id);
```
Comment (simeng-li) [inline packages/core/src/routes/applications/application.ts:170]: Should update the API's response status guard to include the new added error status codes?

File: packages/core/src/tenants/SystemContext.ts [hunk 3]
```diff
@@ -34,6 +37,13 @@ export default class SystemContext {
           hostnameProviderDataGuard
         );
       })(),
+      (async () => {
```
Comment (simeng-li) [inline packages/core/src/tenants/SystemContext.ts:40]: Wondering,  whether we should block the user flow if any of the provided listed here throws an error.

Comment (wangsijie) [reply packages/core/src/tenants/SystemContext.ts:40]: We will block the whole Logto app from start if the config is not empty and has errors.

File: packages/core/src/routes/applications/application.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/routes/applications/application.ts:170]: updated

File: packages/core/src/libraries/protected-app.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/libraries/protected-app.ts:16]: updated

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**